src/qbe_ir_generator.cpp

```diff
@@ -822,14 +822,49 @@ void QbeIrGenerator::Visit(const BinaryExprNode& bin_expr) {
   int left_num = num_recorder.NumOfPrevExpr();
   bin_expr.rhs->Accept(*this);
   int right_num = num_recorder.NumOfPrevExpr();
-  int num = NextLocalNum();
-  // TODO: use the correct instruction for specific data type:
-  // 1. signed or unsigned: currently only supports signed integers.
-  // 2. QBE base data type 'w' | 'l' | 's' | 'd': currently only supports 'w'.
-  WriteInstr_("{} =w {} {}, {}", FuncScopeTemp{num},
-              GetBinaryOperator(bin_expr.op), FuncScopeTemp{left_num},
-              FuncScopeTemp{right_num});
-  num_recorder.Record(num);
+  // The && operator shall yield 1 if both of its operands compare unequal to
+  // 0; otherwise, it yields 0; The || operator shall yield 1 if either of its
+  // operands compare unequal to 0; otherwise, it yields 0.
+  if (bin_expr.op == BinaryOperator::kLand ||
+      bin_expr.op == BinaryOperator::kLor) {
+    const int label_num = NextLabelNum();
+    auto rhs_label = BlockLabel{"logic_rhs", label_num};
+    // Early exit after evaluating the first operand.
+    auto exit_label = BlockLabel{"logic_exit", label_num};
```

Comment:
The early exit is commonly referred to as [short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation). How about we name the label accordingly to make it more easily understood?

```suggestion
    // Early exit after evaluating the first operand.
    auto short_circuit_label = BlockLabel{"short_circuit", label_num};
```